chain/chain/src/chain.rs

```diff
@@ -1876,18 +1876,39 @@ impl Chain {
                 }
             })
             .expect("results should resolve to a final outcome");
-        let receipts = outcomes.split_off(1);
-        let transaction = self
+        let receipts_outcome = outcomes.split_off(1);
+        let transaction: SignedTransactionView = self
             .store
             .get_transaction(hash)?
             .ok_or_else(|| ErrorKind::DBNotFoundErr(format!("Transaction {} is not found", hash)))?
             .clone()
             .into();
+        let transaction_outcome = outcomes.pop().unwrap();
+        let receipt_id_from_transaction = transaction_outcome.outcome.receipt_ids.get(0).cloned();
+        let is_local_receipt = transaction.signer_id == transaction.receiver_id;
+
+        let receipts = receipts_outcome
+            .iter()
+            .filter_map(|outcome| {
+                if Some(outcome.id) == receipt_id_from_transaction && is_local_receipt {
+                    None
```

Comment:
We should expose a method to construct a local receipt from the signed transaction. This will make things easier for end users.

Reply:
but you need to expose the config and the config can change between different protocol versions